client/battle/CBattleInterface.cpp

```diff
@@ -2163,7 +2163,14 @@ void CBattleInterface::handleHex(BattleHex myNumber, int eventType)
 				break;
 			case RISE_DEMONS:
 				if (shere && ourStack && !shere->alive())
-					legalAction = true;
+				{
+					if(!(shere->hasBonusOfType(Bonus::UNDEAD) 
+						|| shere->hasBonusOfType(Bonus::NON_LIVING) 
+						|| vstd::contains(shere->state, EBattleStackState::SUMMONED)
+						|| vstd::contains(shere->state, EBattleStackState::CLONED)
+						))
+						legalAction = true;
+				}					

```

Comment:
BTW is there any particular reason why RISE_DEMONS need it's own code at all?
I just see that Archangel resurrection somehow handled by ANY_CREATURE and it's working fine.